src/LaborMarket/LaborMarket.sol

```diff
@@ -231,38 +162,46 @@ contract LaborMarket is
             "LaborMarket::signal: Already signaled."
         );
 
-        // Lock reputation here
-        uint256 signalAmt = 1;
+        uint256 signalStake = reputationModule.getSignalStake(address(this));
+
+        _lockReputation(
+            msg.sender,
+            signalStake
+        );
 
         submissionSignals[requestId][msg.sender] = true;
 
         unchecked {
             ++signalCount[requestId];
         }
 
-        emit RequestSignal(msg.sender, requestId, signalAmt);
+        emit RequestSignal(msg.sender, requestId, signalStake);
     }
 
     /**
      * @notice Signals interest in reviewing a submission.
-     * @param submissionId The id of the service providers submission.
+     * @param requestId The id of the service providers submission.
      */
-    function signalReview(uint256 submissionId) external onlyMaintainer {
+    function signalReview(uint256 requestId) external onlyMaintainer {
         require(
-            submissionId <= serviceSubmissionId,
-            "LaborMarket::signalReview: Submission does not exist."
+            requestId <= serviceRequestId,
```

Comment:
Looking for opinion: Does the primitive care if the request exists? Why are they submitting a request that doesn't exist?

Reply:
I'm okay with this being taken out. Request IDs are not readily surfaced so if a non-existent request ID is called during the signal, that's an issue at the app layer.

Reply:
Agreed